### DISCUSSION TITLE
Make all searches behave the same way.

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/frontend#12304] [Dashboard]
after initial issue at https://github.com/home-assistant/frontend/issues/11938 and discussion in the community thread, https://community.home-assistant.io/t/2022-4-groups-groups-groups/408965/579

please consider this FR/Discussion;

**Make all searches behave the same way**, i.e. filter based on name or entity_id.
When the result is selected from the searched list. It will show both the name and entity_id. Just like it does in the dropdown list when selecting.
The visible name will continue to be the searchbox, where the entity_id will be static text that is not alterable. However you can highlight and select it.

there are many examples in the linked issue. no need to repeat here for the opening post. 

thanks

**Comment by Petro31** [comment]
My recommendation:

1.  Make all searches behave the same way. I.e. filter based on name or entity_id.
2. When the result is selected from the searched list. It will show both the name and entity_id. Just like it does in the dropdown list when selecting.
2.  The entity name (friendly name) will continue to be the search box (text ctrl), where the entity_id will be static label that is not alterable below the search box.  The static label with the entity_id will be smaller text than the search box.  You'll be able to select the static label so you can copy/paste the entity_id into your yaml.

**Reply by emufan** [reply]
I like all three recommendations! And yes, it's clear, that this has to be improved (back) again.

**Reply by RoboMagus** [reply]
I like your suggestions, but I do have a problem with 3.
Mostly I'm already looking for entity_ids when using the search option, e.g. start typing `sensor.`... Then when clicking on the entity I'm interested in it currently defaults to the entity friendly name. So when you then start typing again you're basically no longer in entity_ids mode, but start editing the friendly_name.

My believe is that as you're looking for specific entities, their only identifier is the entity_id and thus that should be the basis of the search box. Having the friendly name there is nice, but it should definitely be secondary.

**Reply by ristomatti** [reply]
I personally find it bad UX to not show the user an indication what their search is matching into. I also search with an entity ID in mind. For each room I've got a lot of entities of which friendly name begins with the room name, so being able to quickly filter to only a specific device class (e.g. `binary_sensor`) makes searching a lot faster.

As the entity id is anyway in the background, consider making it an advanced (default off) for feature to toggle on a user basis. Otherwise I see things moving into people needing to start renaming their entities to have unique names they really don't wouldn't want to see on their dashboards. There is no unique constraint on the friendly name. I see it as very bad UX to allow a user ending up to a situation theyll have two or more items with the exact same name.

Why make the entity id something you try to hide whenever possible? I believe it to be quite frankly underestimating the end users' intelligence. Even non techy people grow into using unique id's in their day to day life e.g. social security number, phone number, email address, user account and understand why they are needed. Why make entity id like its something scary? It's a lot more friendly than for example a MAC address and in most cases you can even name it like you like. 

<sub>Edit: fixed a nonsensical sentence written in hurry</sub>

**Reply by emufan** [reply]
Esp. as this this config and not enduser dashboard. And even if you really need to have enduser/admin/pro-admin, you can distinguish between these types and use the existing “advanced mode” toggle. Let the admin choose.
